### PR TITLE
feat(backend)(game): add reset-draft endpoint and draft reset flow

### LINKED ISSUE #262: feat(backend)(game): add reset draft endpoint service logic
## Description

Describe the feature briefly.

## Acceptance Criteria

- [ ]

## Notes

Additional context, screenshots, or links.

### LINKED ISSUE #263: feat(backend)(game): add reset draft from confirmed state logic
## Description

Describe the feature briefly.

## Acceptance Criteria

- [ ]

## Notes

Additional context, screenshots, or links.

### LINKED ISSUE #339: feat(backend)(game)(rest-api): REST API foundation (parent issue)
## Description

Describe the feature briefly.

## Acceptance Criteria

- [ ]

## Notes

Additional context, screenshots, or links.

### LINKED ISSUE #345: feat(backend)(game)(rest-api): add reset draft endpoint
## Description

Describe the feature briefly.

## Acceptance Criteria

- [ ]

## Notes

Additional context, screenshots, or links.

### LINKED ISSUE #376: test(backend)(game): add reset draft service tests
## Description

Describe the feature briefly.

## Acceptance Criteria

- [ ]

## Notes

Additional context, screenshots, or links.

### PR DESCRIPTION
**Pull request by miri2468**
## Summary

This PR implements the missing reset-draft feature behind:

```text
POST /api/games/{gameId}/reset-draft
```

The goal is to let the active player discard their current in-progress draft
changes and rebuild the live draft from the authoritative confirmed game state.

This closes the remaining gap in the live draft lifecycle between:

- draft creation
- draft updates
- draw-tile draft mutation
- end-turn draft submission

## Issues

- Closes #376 
- Closes #262 
- Closes #263 
- Closes #345 
- Closes #339 
- Closes #480 
- Closes #474 

## What this PR adds

- `POST /api/games/{gameId}/reset-draft`
- service-layer reset orchestration in `TurnDraftService`
- rebuilding the draft from confirmed game board and confirmed player rack
- clearing draft-only draw markers such as `drawnTile`
- persisting the refreshed draft on the existing entity
- broadcasting `game.draft.updated`
- controller and service tests for the reset flow

## Intended behavior

The reset action should:

1. load the confirmed game
2. load the active draft
3. verify that the acting user is the active player and draft owner
4. rebuild the draft from confirmed server state
5. increment the draft version
6. clear draft-only temporary state
7. save the updated draft
8. broadcast the refreshed draft
9. return `TurnDraftResponse`

## API shape

```kotlin
@PostMapping("/{gameId}/reset-draft")
fun resetDraft(
    @PathVariable gameId: String,
    @RequestHeader("X-User-Id") userId: String
): TurnDraftResponse
```

Expected responses:

- `200 OK` with `TurnDraftResponse`
- `404 NOT_FOUND` if game or active draft is missing
- `409 CONFLICT` if the user is not the current player or does not own the draft

## Important scope boundary

This PR does **not** change:

- confirmed game state
- rule validation
- end-turn commit logic
- score handling
- turn progression

It is only a live-draft reset feature.

## Main implementation location

Most of the logic belongs in:

- [`TurnDraftService.kt`](../apps/backend/src/main/kotlin/at/se2group/backend/service/TurnDraftService.kt)

Related backend context:

- [`GameController.kt`](../apps/backend/src/main/kotlin/at/se2group/backend/api/GameController.kt)
- [`DrawTileService.kt`](../apps/backend/src/main/kotlin/at/se2group/backend/service/DrawTileService.kt)
- [`EndTurnService.kt`](../apps/backend/src/main/kotlin/at/se2group/backend/service/EndTurnService.kt)
- [`docs/pr-reset-draft.md`](./pr-reset-draft.md)

### CODE REVIEW

File: apps/backend/src/main/kotlin/at/se2group/backend/service/TurnDraftService.kt [hunk 1]
```diff
@@ -126,4 +127,55 @@ class TurnDraftService(
...
+       return savedDraft
+
+    }
+
+    private fun createDraftFromConfirmedGame(
+        game: ConfirmedGame,
+        playerUserId: String,
+        version: Long
+    ): TurnDraft {
+        val activePlayer = game.players.first { it.userId == playerUserId }
```
Comment (kingjulien1) [inline apps/backend/src/main/kotlin/at/se2group/backend/service/TurnDraftService.kt]: Would it be safer to use `firstOrNull` here and throw a more explicit domain error if the player is missing?`first { ... }` will raise a generic `NoSuchElementException`, which makes inconsistent game-state failures harder to understand if this assumption is ever broken.

```kotlin
val activePlayer = game.players.firstOrNull { it.userId == playerUserId }
    ?: throw IllegalArgumentException("Player $playerUserId is not part of game ${game.id}")
```

If the project already uses custom exceptions, then something like this would be nicer:

```kotlin
val activePlayer = game.players.firstOrNull { it.userId == playerUserId }
    ?: throw GameNotFoundException("Player $playerUserId is not part of game ${game.id}")
```

File: apps/backend/src/test/kotlin/at/se2group/backend/api/GameControllerTest.kt [hunk 2]
```diff
@@ -402,4 +402,91 @@ class GameControllerTest {
...
+        }.andExpect {
+            status { isOk() }
+            jsonPath("$.gameId") { value("game-1") }
+            jsonPath("$.playerUserId") { value("user-1") }
+            jsonPath("$.version") { value("4") }
+        }
+    }
+
+    @Test
+    fun `resetDraft returns 400 when game not found`() {
```
Comment (kingjulien1) [inline apps/backend/src/test/kotlin/at/se2group/backend/api/GameControllerTest.kt]: The test name says `400`, but the assertion below checks `isNotFound()` / `404`. Could we rename the test to returns `404` when game not found so the intent matches the expectation?

**Review by kingjulien1 — CHANGES_REQUESTED** [review]
Nice addition overall 👍  — the endpoint shape is clear, the service flow is easy to follow, and the test coverage for success, not-found, and conflict cases looks good. I only left a couple of small follow-ups around one service edge case and a few naming/style nits.

**Review by kingjulien1 — APPROVED** [review]
thanks for resolving my review 👍 You can merge now :)
